libraries/ui/src/ui/types/ContextAwareProfile.h

```diff
@@ -47,10 +49,19 @@ class ContextAwareProfile : public ContextAwareProfileParent {
     };
 
     ContextAwareProfile(QQmlContext* parent);
+    ~ContextAwareProfile();
+
+private:
+    typedef QSet<ContextAwareProfile*> ContextAwareProfileSet;
+    typedef QHash<QQmlContext*, ContextAwareProfileSet> ContextMap;
+
     QQmlContext* _context{ nullptr };
-    bool _cachedValue{ false };
-    quint64 _cacheExpiry{ 0 };
-    constexpr static quint64 MAX_CACHE_AGE = MSECS_PER_SECOND;
+    std::atomic<bool> _isRestricted{ false };
+
+    static QReadWriteLock _global_contextMapProtect;
+    static ContextMap _global_contextMap;
```

Comment:
eh I’m still not sure about these.  we don’t use the _global_ prefix anywhere else.  I’d settle for _globalContextMap and _globalContextMapProtect though?

Reply:
So there is nothing in the coding standard that permits global variables to be identified as such? It seems like that's very similar to the reason why all member variables must start with _